python/cudf_polars/cudf_polars/dsl/expr.py

```diff
@@ -667,36 +667,39 @@ def do_evaluate(
         mapping: Mapping[Expr, Column] | None = None,
     ) -> Column:
         """Evaluate this expression given a dataframe for context."""
-        columns = [
-            child.evaluate(df, context=context, mapping=mapping)
-            for child in self.children
-        ]
-        if self.name == pl_expr.StringFunction.Lowercase:
-            (column,) = columns
-            return Column(plc.strings.case.to_lower(column.obj))
-        elif self.name == pl_expr.StringFunction.Uppercase:
-            (column,) = columns
-            return Column(plc.strings.case.to_upper(column.obj))
-        elif self.name == pl_expr.StringFunction.EndsWith:
-            column, suffix = columns
-            return Column(plc.strings.find.ends_with(column.obj, suffix.obj))
-        elif self.name == pl_expr.StringFunction.StartsWith:
-            column, suffix = columns
-            return Column(plc.strings.find.starts_with(column.obj, suffix.obj))
-        elif self.name == pl_expr.StringFunction.Contains:
-            column, pattern = columns
+        if self.name == pl_expr.StringFunction.Contains:
+            child, pattern = self.children
+            column = child.evaluate(df, context=context, mapping=mapping)
+            assert isinstance(pattern, Literal)
+
             literal, _ = self.options
             if literal:
-                return Column(plc.strings.find.contains(column.obj, pattern.obj))
+                return Column(plc.strings.find.contains(column.obj, pattern.value))
```

Comment:
for literal contains, can we support columns as patterns? libcudf does: https://docs.rapids.ai/api/libcudf/stable/group__strings__find#ga9acbc587765007c5e35811c07e990b03

Reply:
supported this in https://github.com/rapidsai/cudf/pull/15918/commits/e45fbed78d189a0fc7aa261200cfdfc462b0fe17